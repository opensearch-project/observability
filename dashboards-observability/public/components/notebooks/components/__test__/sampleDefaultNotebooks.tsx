/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

// Sample notebook with all input and output
export const sampleNotebook1 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T18:37:56.844Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [{ result: '# Type your input here', outputType: 'MARKDOWN', execution_time: '0s' }],
      input: { inputText: '# Type your input here', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T18:00:59.845Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
    {
      output: [{ result: '# add code here\n', outputType: 'MARKDOWN', execution_time: '0s' }],
      input: { inputText: '# add code here\n', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:01:07.662Z',
      dateModified: '2020-08-20T18:01:18.295Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
    {
      output: [{ result: '', outputType: 'VISUALIZATION', execution_time: '0s' }],
      input: {
        inputText:
          '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
        inputType: 'VISUALIZATION',
      },
      dateCreated: '2020-08-20T18:37:44.809Z',
      dateModified: '2020-08-20T18:37:56.844Z',
      id: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    },
  ],
};

// Parsed Output of sample notebook1
export const sampleParsedParagraghs1 = [
  {
    uniqueId: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 1,
    inp: '# Type your input here',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: ['# Type your input here'],
  },
  {
    uniqueId: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 2,
    inp: '# add code here\n',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: ['# add code here\n'],
  },
  {
    uniqueId: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: true,
    vizObjectInput:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    id: 3,
    inp:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    lang: 'text/x-',
    editorLanguage: '',
    typeOut: ['VISUALIZATION'],
    out: [''],
  },
];

// Sample notebook with all input and cleared outputs
export const sampleNotebook2 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      input: {
        inputText:
          '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
        inputType: 'MARKDOWN',
      },
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
    {
      output: [],
      input: { inputText: '# no output here\n', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:01:07.662Z',
      dateModified: '2020-08-20T19:09:19.210Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
    {
      output: [],
      input: {
        inputText:
          '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
        inputType: 'VISUALIZATION',
      },
      dateCreated: '2020-08-20T18:37:44.809Z',
      dateModified: '2020-08-20T18:37:56.844Z',
      id: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    },
  ],
};

// Parsed Output of sample notebook2
export const sampleParsedParagraghs2 = [
  {
    uniqueId: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 1,
    inp:
      '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: [
      '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
    ],
  },
  {
    uniqueId: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 2,
    inp: '# no output here\n',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: [],
    out: [],
  },
  {
    uniqueId: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: true,
    vizObjectInput:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    id: 3,
    inp:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    lang: 'text/x-',
    editorLanguage: '',
    typeOut: [],
    out: [],
  },
];

// Sample notebook with no input
export const sampleNotebook3 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
  ],
};

// Sample notebook with no paragraph id
export const sampleNotebook4 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
    },
  ],
};

// Sample notebook with no input and output
export const sampleNotebook5 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
  ],
};
