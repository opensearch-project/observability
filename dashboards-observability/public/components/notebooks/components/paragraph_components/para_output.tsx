/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiCodeBlock, EuiSpacer, EuiText } from '@elastic/eui';
import MarkdownRender from '@nteract/markdown';
import { Media } from '@nteract/outputs';
import moment from 'moment';
import React, { useState } from 'react';
import {
  DashboardContainerInput,
  DashboardStart,
} from '../../../../../../../src/plugins/dashboard/public';
import { ParaType } from '../../../../../common/types/notebooks';
import { uiSettingsService } from '../../../../../common/utils';
import { QueryDataGridMemo } from './para_query_grid';

/*
 * "ParaOutput" component is used by notebook to populate paragraph outputs for an open notebook.
 *
 * Props taken in as params are:
 * para - parsed paragraph from notebook
 *
 * Outputs component of nteract used as a container for notebook UI.
 * https://components.nteract.io/#outputs
 */
export const ParaOutput = (props: {
  para: ParaType;
  visInput: DashboardContainerInput;
  setVisInput: (input: DashboardContainerInput) => void;
  DashboardContainerByValueRenderer: DashboardStart['DashboardContainerByValueRenderer'];
}) => {
  const createQueryColumns = (jsonColumns: any[]) => {
    let index = 0;
    let datagridColumns = [];
    for (index = 0; index < jsonColumns.length; ++index) {
      const datagridColumnObject = {
        id: jsonColumns[index].name,
        displayAsText: jsonColumns[index].name,
      };
      datagridColumns.push(datagridColumnObject);
    }
    return datagridColumns;
  };

  const getQueryOutputData = (queryObject: any) => {
    const data = [];
    let index = 0;
    let schemaIndex = 0;
    for (index = 0; index < queryObject.datarows.length; ++index) {
      let datarowValue = {};
      for (schemaIndex = 0; schemaIndex < queryObject.schema.length; ++schemaIndex) {
        const columnName = queryObject.schema[schemaIndex].name;
        if (typeof queryObject.datarows[index][schemaIndex] === 'object') {
          datarowValue[columnName] = JSON.stringify(queryObject.datarows[index][schemaIndex]);
        } else if (typeof queryObject.datarows[index][schemaIndex] === 'boolean') {
          datarowValue[columnName] = queryObject.datarows[index][schemaIndex].toString();
        } else {
          datarowValue[columnName] = queryObject.datarows[index][schemaIndex];
        }
      }
      data.push(datarowValue);
    }
    return data;
  };

  const outputBody = (key: string, typeOut: string, val: string) => {
    /* Returns a component to render paragraph outputs using the para.typeOut property
     * Currently supports HTML, TABLE, IMG
     * TODO: add table rendering
     */

    if (typeOut !== undefined) {
      switch (typeOut) {
        case 'QUERY':
          const inputQuery = para.inp.substring(4, para.inp.length);
          const queryObject = JSON.parse(val);
          if (queryObject.hasOwnProperty('error')) {
            return <EuiCodeBlock key={key}>{val}</EuiCodeBlock>;
          } else {
            const columns = createQueryColumns(queryObject.schema);
            const data = getQueryOutputData(queryObject);
            const [visibleColumns, setVisibleColumns] = useState(() => columns.map(({ id }) => id));
            return (
              <div>
                <EuiText key={'query-input-key'}>
                  <b>{inputQuery}</b>
                </EuiText>
                <EuiSpacer />
                <QueryDataGridMemo
                  key={key}
                  rowCount={queryObject.datarows.length}
                  queryColumns={columns}
                  visibleColumns={visibleColumns}
                  setVisibleColumns={setVisibleColumns}
                  dataValues={data}
                />
              </div>
            );
          }
        case 'MARKDOWN':
          return (
            <EuiText key={key} className="markdown-output-text">
              <MarkdownRender source={val} />
            </EuiText>
          );
        case 'VISUALIZATION':
          const dateFormat = uiSettingsService.get('dateFormat');
          let from = moment(visInput?.timeRange?.from).format(dateFormat);
          let to = moment(visInput?.timeRange?.to).format(dateFormat);
          from = from === 'Invalid date' ? visInput.timeRange.from : from;
          to = to === 'Invalid date' ? visInput.timeRange.to : to;
          return (
            <>
              <EuiText size="s" style={{ marginLeft: 9 }}>
                {`${from} - ${to}`}
              </EuiText>
              <DashboardContainerByValueRenderer
                key={key}
                input={visInput}
                onInputUpdated={setVisInput}
              />
            </>
          );
        case 'HTML':
          return (
            <EuiText key={key}>
              <Media.HTML data={val} />
            </EuiText>
          );
        case 'TABLE':
          return <pre key={key}>{val}</pre>;
        case 'IMG':
          return <img alt="" src={'data:image/gif;base64,' + val} key={key} />;
        default:
          return <pre key={key}>{val}</pre>;
      }
    } else {
      console.log('output not supported', typeOut);
      return <pre />;
    }
  };

  const { para, DashboardContainerByValueRenderer, visInput, setVisInput } = props;

  return (
    !para.isOutputHidden ? (
      <>
        {para.typeOut.map((typeOut: string, tIdx: number) => {
          return outputBody(para.uniqueId + '_paraOutputBody', typeOut, para.out[tIdx])
        }
        )}
      </>
    ) : null
  );
};
