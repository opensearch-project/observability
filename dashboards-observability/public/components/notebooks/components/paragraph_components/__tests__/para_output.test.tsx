/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { sampleParsedParagraghs1 } from '../../helpers/__tests__/sampleDefaultNotebooks';
import { ParaOutput } from '../para_output';

describe('<ParaOutput /> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders markdown outputs', () => {
    const para = sampleParsedParagraghs1[0];
    para.isSelected = true;
    const setVisInput = jest.fn();
    const utils = render(
      <ParaOutput
        key={para.uniqueId}
        para={para}
        visInput={jest.fn()}
        setVisInput={setVisInput}
        DashboardContainerByValueRenderer={jest.fn()}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders query outputs', () => {
    const para = sampleParsedParagraghs1[3];
    para.isSelected = true;
    const setVisInput = jest.fn();
    const utils = render(
      <ParaOutput
        key={para.uniqueId}
        para={para}
        visInput={jest.fn()}
        setVisInput={setVisInput}
        DashboardContainerByValueRenderer={jest.fn()}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders visualization outputs', () => {
    const para = sampleParsedParagraghs1[2];
    para.isSelected = true;
    const setVisInput = jest.fn();
    const utils = render(
      <ParaOutput
        key={para.uniqueId}
        para={para}
        visInput={{
          timeRange: { from: '2020-07-21T18:37:44.710Z', to: '2020-08-20T18:37:44.710Z' },
        }}
        setVisInput={setVisInput}
        DashboardContainerByValueRenderer={() => null}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders other types of outputs', () => {
    const para = sampleParsedParagraghs1[0];
    para.isSelected = true;
    para.typeOut = ['HTML', 'TABLE', 'IMG', 'UNKNOWN', undefined];
    para.out = ['', '', '', '', ''];
    const setVisInput = jest.fn();
    const utils = render(
      <ParaOutput
        key={para.uniqueId}
        para={para}
        visInput={jest.fn()}
        setVisInput={setVisInput}
        DashboardContainerByValueRenderer={jest.fn()}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
