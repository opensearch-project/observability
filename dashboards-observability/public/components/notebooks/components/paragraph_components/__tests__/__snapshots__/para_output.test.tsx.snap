// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParaOutput /> spec renders markdown outputs 1`] = `
<div
  class="euiText euiText--medium markdown-output-text"
>
  <div
    class="markdown-body "
  >
    <h1>
      Type your input here
    </h1>
  </div>
</div>
`;

exports[`<ParaOutput /> spec renders other types of outputs 1`] = `
<div
  class="euiText euiText--medium"
>
  <div
    class="sc-crXcEl gqVnJo"
  />
</div>
`;

exports[`<ParaOutput /> spec renders query outputs 1`] = `
<div>
  <div
    class="euiText euiText--medium"
  >
    <b>
       select * from opensearch_dashboards_sample_data_flights limit 2
    </b>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    id="queryDataGrid"
  >
    <span
      class="euiLoadingSpinner euiLoadingSpinner--xLarge"
    />
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
    <div
      class="euiDataGrid__focusWrap"
      data-focus-lock-disabled="disabled"
    >
      <div
        class="euiDataGrid euiDataGrid--bordersAll euiDataGrid--headerShade euiDataGrid--footerOverline euiDataGrid--rowHoverHighlight euiDataGrid--stickyFooter"
      />
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
  </div>
</div>
`;

exports[`<ParaOutput /> spec renders visualization outputs 1`] = `
<div
  class="euiText euiText--small"
  style="margin-left: 9px;"
>
  2020-07-21T18:37:44+00:00 - 2020-08-20T18:37:44+00:00
</div>
`;
